### DISCUSSION TITLE
Is it possible to build the PhysX in a wsl2 system that uses XMing for the GUI?

### DISCUSSION THREAD

**Discussion by shutengW** [NVIDIA-Omniverse/PhysX#150] [Q&A]
Hi,

I am new to PhysX and I wonder if it is possible to build it in my WSL2?

I tried it and the following error pops:

./generate_projects.sh
Preset parameter required, available presets:
(0) linux-aarch64 <--- Linux-aarch64 gcc PhysX SDK general settings
(1) linux <--- Linux clang PhysX SDK general settings
Enter preset number: 0
Running generate_projects.sh linux-aarch64
Running packman in preparation for cmake ...
Processing project file '/home/shuteng/PhysX/physx/dependencies.xml'
Using preset xml: buildtools/presets/public/linux-aarch64.xml
Target platform: linuxAarch64 using compiler: gcc
PM_CMakeModules_PATH: /home/shuteng/packman-repo/chk/CMakeModules/1.28.trunk.32494385
PM_PATHS: /home/shuteng/packman-repo/chk/CMakeModules/1.28.trunk.32494385;/home/shuteng/packman-repo/chk/clang-physxmetadata/4.0.0.32489833_1;/home/shuteng/packman-repo/chk/PhysXGpu/104.2-5.1.264.32487460-public;/home/shuteng/packman-repo/chk/rapidjson/1.1.0-67fac85-073453e1
Cmake: cmake
Not searching for unused variables given on the command line.
CMake Error at /home/shuteng/packman-repo/chk/CMakeModules/1.28.trunk.32494385/linux/LinuxAarch64.cmake:14 (message):
  Failed to find aarch64-linux-gnu-gcc
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/CMakeDetermineSystem.cmake:93 (include)
  CMakeLists.txt:33 (project)

CMake Error: CMake was unable to find a build program corresponding to "Unix Makefiles".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
-- Configuring incomplete, errors occurred!
Not searching for unused variables given on the command line.
CMake Error at /home/shuteng/packman-repo/chk/CMakeModules/1.28.trunk.32494385/linux/LinuxAarch64.cmake:14 (message):
  Failed to find aarch64-linux-gnu-gcc
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/CMakeDetermineSystem.cmake:93 (include)
  CMakeLists.txt:33 (project)

CMake Error: CMake was unable to find a build program corresponding to "Unix Makefiles".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
-- Configuring incomplete, errors occurred!
Not searching for unused variables given on the command line.
CMake Error at /home/shuteng/packman-repo/chk/CMakeModules/1.28.trunk.32494385/linux/LinuxAarch64.cmake:14 (message):
  Failed to find aarch64-linux-gnu-gcc
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/CMakeDetermineSystem.cmake:93 (include)
  CMakeLists.txt:33 (project)

CMake Error: CMake was unable to find a build program corresponding to "Unix Makefiles".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
-- Configuring incomplete, errors occurred!
Not searching for unused variables given on the command line.
CMake Error at /home/shuteng/packman-repo/chk/CMakeModules/1.28.trunk.32494385/linux/LinuxAarch64.cmake:14 (message):
  Failed to find aarch64-linux-gnu-gcc
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/CMakeDetermineSystem.cmake:93 (include)
  CMakeLists.txt:33 (project)

CMake Error: CMake was unable to find a build program corresponding to "Unix Makefiles".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
-- Configuring incomplete, errors occurred!

**Comment by preist-nvidia** [comment]
Hi @shutengW did you look at our platform readme already? https://github.com/NVIDIA-Omniverse/PhysX/blob/release/104.2/physx/documentation/platformreadme/linux/README_LINUX.md

You will want to use preset 1 as well I would assume.

**Reply by shutengW** [reply]
Hi, Yes, I did have a look at it. I am pretty sure all the libs are installed. I did have used preset 1, and the error remains as shown below. And I am pretty sure all the packages are properly installed. So I am a little confused whether this could work out in my WSL2. Any hints would be really appreciated.

./generate_projects.sh
Preset parameter required, available presets:
(0) linux-aarch64 <--- Linux-aarch64 gcc PhysX SDK general settings
(1) linux <--- Linux clang PhysX SDK general settings
Enter preset number: 1
Running generate_projects.sh linux
Running packman in preparation for cmake ...
Processing project file '/home/shuteng/PhysX/physx/dependencies.xml'
Using preset xml: buildtools/presets/public/linux.xml
Target platform: linux using compiler: clang
PM_CMakeModules_PATH: /home/shuteng/packman-repo/chk/CMakeModules/1.28.trunk.32494385
PM_PATHS: /home/shuteng/packman-repo/chk/CMakeModules/1.28.trunk.32494385;/home/shuteng/packman-repo/chk/clang-physxmetadata/4.0.0.32489833_1;/home/shuteng/packman-repo/chk/PhysXGpu/104.2-5.1.264.32487460-public;/home/shuteng/packman-repo/chk/opengl-linux/2017.5.19.1;/home/shuteng/packman-repo/chk/rapidjson/1.1.0-67fac85-073453e1
Cmake: cmake
Not searching for unused variables given on the command line.
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:33 (project):
  The CMAKE_C_COMPILER:

    clang

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
  the compiler, or to the compiler name if it is in the PATH.


CMake Error at CMakeLists.txt:33 (project):
  The CMAKE_CXX_COMPILER:

    clang++

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.


-- Configuring incomplete, errors occurred!
See also "/home/shuteng/PhysX/physx/compiler/linux-debug/CMakeFiles/CMakeOutput.log".
See also "/home/shuteng/PhysX/physx/compiler/linux-debug/CMakeFiles/CMakeError.log".
Not searching for unused variables given on the command line.
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:33 (project):
  The CMAKE_C_COMPILER:

    clang

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
  the compiler, or to the compiler name if it is in the PATH.


CMake Error at CMakeLists.txt:33 (project):
  The CMAKE_CXX_COMPILER:

    clang++

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.


-- Configuring incomplete, errors occurred!
See also "/home/shuteng/PhysX/physx/compiler/linux-checked/CMakeFiles/CMakeOutput.log".
See also "/home/shuteng/PhysX/physx/compiler/linux-checked/CMakeFiles/CMakeError.log".
Not searching for unused variables given on the command line.
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:33 (project):
  The CMAKE_C_COMPILER:

    clang

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
  the compiler, or to the compiler name if it is in the PATH.


CMake Error at CMakeLists.txt:33 (project):
  The CMAKE_CXX_COMPILER:

    clang++

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.


-- Configuring incomplete, errors occurred!
See also "/home/shuteng/PhysX/physx/compiler/linux-profile/CMakeFiles/CMakeOutput.log".
See also "/home/shuteng/PhysX/physx/compiler/linux-profile/CMakeFiles/CMakeError.log".
Not searching for unused variables given on the command line.
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:33 (project):
  The CMAKE_C_COMPILER:

    clang

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
  the compiler, or to the compiler name if it is in the PATH.


CMake Error at CMakeLists.txt:33 (project):
  The CMAKE_CXX_COMPILER:

    clang++

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.


-- Configuring incomplete, errors occurred!
See also "/home/shuteng/PhysX/physx/compiler/linux-release/CMakeFiles/CMakeOutput.log".
See also "/home/shuteng/PhysX/physx/compiler/linux-release/CMakeFiles/CMakeError.log".

**Reply by shutengW** [reply]
and fyi I installed gcc in my Ubuntu. I did not use clang..

gcc --version
gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

**Reply by shutengW** [reply]
It works by installing the clang compiler. I have figured this out. Thx. Although I am still not sure why gcc does not work...

**Reply by preist-nvidia** [reply]
Ok good, but yes I agree that gcc should also work.
